src/coreclr/src/gc/gc.cpp

```diff
@@ -2088,8 +2088,9 @@ uint8_t* tree_search (uint8_t* tree, uint8_t* old_address);
 #ifdef USE_VXSORT
 void do_vxsort(uint8_t** low, uint8_t** high, unsigned int depth)
 {
-    assert(SupportsInstructionSet(InstructionSet::AVX2));
-    if (SupportsInstructionSet(InstructionSet::AVX512F))
+    assert(IsSupportedInstructionSet(InstructionSet::AVX2));
+    // use AVX512F only if the list is large enough to pay for downclocking impact
+    if (IsSupportedInstructionSet(InstructionSet::AVX512F) && ((high -low) > 128*1024))
```

Comment:
The thresholds `128*1024` are used twice for AVX512F and `8*1024` for AVX2; maybe take them to consts? (or GC config values?)

Reply:
Done - thanks!